packages/components/basic-table/types/basic-table.d.ts

```diff
@@ -1,6 +1,29 @@
-export declare class TSBasicTable {
-	dir?: 'rtl' | 'ltr' | 'auto';
-	cols: any[];
-	data: any[];
-	selectedIds?: any[];
+export interface TSBasicTableHTMLAttributes {
+	/**  Direction of the component 'rtl' or 'ltr'  */
+	dir?: string;
+
+	/**  <br> List of columns configs, including: <br> - property: Property key of the column in data object, <br> - value: value of the title of column', <br> - visibility?: Which screen sizes this column should be visible -> 'always-visible'(default) or 'desktop-only' or 'mobile-only', <br> - size?: 'small' or 'medium' or 'large', <br> - display?: 'left' or 'right' or 'center', <br> - renderer?: you can pass a renderer function to customize the content of the cells in this column, args: (cellValue, rowObject) <br>  */
+	cols?: [];
```

Comment:
Is there a reason that we have `cols` and `data` as optional?

Reply:
No, I just don't know how to decide is it a required or optional property. Probably we can make all attributes optional (otherwise it will mess with jsx complaining that developer have to specify all the attributes) and all properties non optional by default.

Reply:
I think for now this is good enough to just have them as optional even though it's not true. For some properties we can know that they are not required, based on being boolean, or having default value, but not a reliable and enough. We should discuss how we should mark them in the code, specially since for some components slots are actually required and we should let developers know that.